src-env (msdb)/_MicroserviceTemplate_.WebApi/Startup.cs

```diff
@@ -0,0 +1,269 @@
+            services.AddScoped<ErrorFilterAttribute>();
+
+            services.AddTransient<IDbMigrationService, _MicroserviceTemplate_DbMigrationService>();
+
+            services.AddTransient<IMyTablesRepository, MyTablesRepository>();
+
+            services.AddTransient<IEmptyRepeatableJobsService, EmptyRepeatableJobsService>();
+
+            services.AddSingleton<IBackgroundTaskQueue, BackgroundTaskQueue>();
+            services.AddHostedService<QueuedHostedService>();
+        }
+
+        /// <summary>
+        /// Adds authorization.
+        /// </summary>
+        /// <param name="services">Services.</param>
+        protected virtual void AddAuthorization(IServiceCollection services)
+        {
+            services.Configure<BenrazAuthorizationAuthGatewaySettings>(x =>
+            {
+                x.BaseUrl = Configuration.GetValue<string>("General:AuthorizationBaseUrl");
+            });
+            services.AddTransient<IBenrazAuthorizationAuthGateway, BenrazAuthorizationAuthGateway>();
+
+            services.Configure<TokenValidationServiceSettings>(Configuration.GetSection("TokenValidation"));
+            services.AddTransient<ITokenValidationService, TokenValidationService>();
+
+            services.AddTransient<TokenValidator>();
+
+            services
+                .AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
+                .AddJwtBearer(options =>
+                {
+                    var serviceProvider = services.BuildServiceProvider();
+                    var tokenValidationService = serviceProvider.GetRequiredService<ITokenValidationService>();
+
+                    // Switch to using TokenHandlers
+                    options.TokenHandlers.Clear();  // Clear existing handlers if any
+                    options.TokenHandlers.Add(new JwtSecurityTokenHandler());  // Add JwtSecurityTokenHandler for handling JWTs
+
+                    // Configure the TokenValidationParameters to specify how tokens should be validated
+                    options.TokenValidationParameters = new TokenValidationParameters
+                    {
+                        ClockSkew = TimeSpan.FromMinutes(5),
+                        IssuerSigningKeyResolver = tokenValidationService.IssuerSigningKeyResolver,
+                        ValidateAudience = true,
+                        AudienceValidator = tokenValidationService.AudienceValidator,
+                        ValidateIssuer = true,
+                        IssuerValidator = tokenValidationService.IssuerValidator,
+
+                        // Include additional validation settings as needed
+                    };
+
+                    options.Audience = Configuration["Jwt:Audience"];
+                    options.Authority = Configuration["Jwt:Authority"];
+
+                });
+
+            services
+                .AddAuthorization(options =>
+                {
+                    options.AddClaimsPolicy(
```

Comment:
you need to add new claim file in the doc folder for this template

Reply:
Added new claim file with the below name
1. Microservice template claims env (msdb).sql
2. Microservice template claims env (msdb).xlsx